js/directives/lucca/user-picker.js

```diff
@@ -0,0 +1,322 @@
+	"<small ng-if=\"showFormerEmployees && user.isFormerEmployee\">VAR_TRAD Parti(e) le {{user.dtContractEnd | luifMoment: 'll'}}</small>" +
+	"<small ng-if=\"user.overflow\">{{user.overflow}}</small>" +
+	"</ui-select-choices>";
+
+	var userPickerTemplate = "<ui-select ng-model=\"user\" theme=\"bootstrap\"" +
+	"class=\"lui regular nguibs-ui-select\" on-select=\"onSelect()\" on-remove=\"onRemove()\" ng-disabled=\"controlDisabled\">" +
+	"<ui-select-match placeholder=\"{{ $select.selected.firstName }} {{$select.selected.lastName}}\">{{ $select.selected.firstName }} {{$select.selected.lastName}}</ui-select-match>" +
+	uiSelectChoicesTemplate +
+	"</ui-select>";
+
+	var userPickerMultipleTemplate = "<ui-select multiple ng-model=\"selected.users\" theme=\"bootstrap\"" +
+	"class=\"lui regular nguibs-ui-select\" on-select=\"addSelectedUser()\" on-remove=\"onRemove()\" ng-disabled=\"controlDisabled\">" +
+	"<ui-select-match placeholder=\"VAR_TRAD Sélectionner un utilisateur...\">{{$item.firstName}} {{$item.lastName}} " +
+	"<span ng-if=\"$item.hasHomonyms\" ng-repeat=\"property in properties\">&lt{{getProperty($item, property)}}&gt</span>" +
+	"<span ng-if=\"$item.isFormerEmployee\">&lt;VAR_TRAD Parti(e) le {{$item.dtContractEnd | luifMoment: 'll'}}&gt;</span>" +
+	"</ui-select-match>" +
+	uiSelectChoicesTemplate +
+	"</ui-select>";
+
+
+	angular.module('lui.directives')
+	.directive('luidUserPicker', function () {
+		return {
+			restrict: 'E',
+			controller: "luidUserPickerController",
+			template: userPickerTemplate,
+			require: "luidUserPicker",
+			scope: {
+				showFormerEmployees: "=",
+				onSelect: "&",
+				onRemove: "&",
+				controlDisabled: "=",
+				useCustomFilter: "=",
+				customFilter: "&",
+				appId: "@",
+				operation: "@",
+				homonymsProperties: "@"
+			},
+			link: function (scope, elt, attrs, ctrls) {
+				if (attrs.homonymsProperties) {
+					scope.properties = attrs.homonymsProperties.split(',');
+				}
+				else {
+					scope.properties = DEFAULT_HOMONYMS_PROPERTIES;
+				}
+			}
+		};
+	})
+
+	.directive('luidUserPickerMultiple', function () {
+		return {
+			restrict: 'E',
+			controller: "luidUserPickerController",
+			template: userPickerMultipleTemplate,
+			require: "luidUserPickerController",
+			scope: {
+				showFormerEmployees: "=",
+				onSelect: "&",
+				onRemove: "&",
+				controlDisabled: "=",
+				useCustomFilter: "=",
+				customFilter: "&",
+				appId: "@",
+				operation: "@",
+				homonymsProperties: "@"
+			},
+			link: function (scope, elt, attrs, ctrls) {
+				if (attrs.homonymsProperties) {
+					scope.properties = attrs.homonymsProperties.split(',');
+				}
+				else {
+					scope.properties = DEFAULT_HOMONYMS_PROPERTIES;
+				}
+			}
+		};
+	})
+
+	.controller("luidUserPickerController", ['$scope', '$http', 'moment', '$timeout', '$q', function ($scope, $http, moment, $timeout, $q) {
+		var timeout = {};
+		var promise;
```

Comment:
`timeout` and `promise`, what are they and what are their purpose. the naming is too generic for me to know what is their use.

also at line ~104 you have this
```js
if (promise) {
  promise.then(function (response) {}); // do nothing
}
```
that does *not* enlighten me

I know it is linked to the way pagination is handled but it is not clear enough for anybody new to this code